### PR TITLE
SVC in C

### PR DESCRIPTION
**Pull request by humleflue**
- Add base logic
- Add CuTest

### CODE REVIEW

File: c-svc/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Support Vector Classifier in C
+
+## Getting started
+To run the SVC started run `make` in your terminal/console and execute the generated svc.out/svc.exe file.
```
Comment (christopher-jensen) [inline c-svc/README.md:4]: "to run the SVC started run" Not sure what you mean?

File: c-svc/svc.c [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#include "transform.h"
+
+int main(void) {
+    return 0;
+}
```
Comment (christopher-jensen) [inline c-svc/svc.c]:
```suggestion
}

```

File: c-svc/transform.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+/* Adds the x, y and z from newVector to the observation array in the tso,
+ * If the tso.observation array is full we do nothing
+ * Note: tso is both input and output param 
+ */
+void addVectorToTimeSeriesObservation(TimeSeriesObservation *tso, double newVector[VECTOR_DIM]) {
+    int i = tso->_currentIndex;
+    
+    if(!tso->isFull) {
+        tso->observation[i  ] = newVector[0];
```
Comment (christopher-jensen) [inline c-svc/transform.c:11]:
```suggestion
        tso->observation[i]   = newVector[0];
```
Maybe we can make a loop 
```
for (j = 0; j < 3; j++){
   tso->observation[i+j] = newVector[j]
}
```

File: c-svc/transform.c [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    double x = 0.0, y = 0.0, z = 0.0;
+
+    for(i = 0; i < TIME_SERIES_OBSERVATION_DIM; i++) {
+        if(i % 3 == 0)
+            x += o.observation[i];
+        else if(i % 3 == 1)
+            y += o.observation[i];
+        else
+            z += o.observation[i];
+    }
```
Comment (christopher-jensen) [inline c-svc/transform.c:37]:
```suggestion
    for(i = 0; i < TIME_SERIES_OBSERVATION_DIM; i+=3) {
        x += o.observation[i];
        y += o.observation[i+1];
        z += o.observation[i+2];
    }
```

File: c-svc/test/TransformTests.c [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	- The name of the method being tested.
+	- The scenario under which it's being tested.
+	- The expected behavior when the scenario is invoked.
+*/
+
+void addVectorToTimeSeriesObservation_tryToStackSmash_shouldNotAffectStack(CuTest* tc)
+{
+	/* Arrange */
+	TimeSeriesObservation o;
+	double newVector[3], arbitraryDouble = 0.42;
```
Comment (christopher-jensen) [inline c-svc/test/TransformTests.c:19]:
```suggestion
	double newVector[VECTOR_DIM], arbitraryDouble = 0.42;
```

File: c-svc/test/TransformTests.c [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+	int i;
+
+	o._currentIndex = 0;
+	newVector[0] = arbitraryDouble;
+	newVector[1] = arbitraryDouble;
+	newVector[2] = arbitraryDouble;
```
Comment (christopher-jensen) [inline c-svc/test/TransformTests.c:25]:
```suggestion
for(i = 0; i < VECTOR_DIM; i++){
	newVector[i] = arbitraryDouble;
}
```

File: c-svc/test/TransformTests.c [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
```
Comment (christopher-jensen) [inline c-svc/test/TransformTests.c:48]: You did this in the function above aswell. Maybe make abstraction function

File: c-svc/test/TransformTests.c [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+	/* Act */
+	for(i = 0; i < TIME_SERIES_OBSERVATION_DIM / 3; i++) {
+		addVectorToTimeSeriesObservation(&o, newVector);
+	}
+
```
Comment (christopher-jensen) [inline c-svc/test/TransformTests.c:54]: Same as before

File: c-svc/test/TransformTests.c [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+	TimeSeriesObservation tso;
+	AggregatedObservation ao;
+	int i;
+	double expected = 3.1;
+
+	/* We fill up the tso's observations with the expected value,
+	   which means that the average will always be the expected value */
+	for(i = 0; i < TIME_SERIES_OBSERVATION_DIM; i++) {
+		tso.observation[i] = expected;
+	}
```
Comment (christopher-jensen) [inline c-svc/test/TransformTests.c:72]: Again using same functionality

**Review by christopher-jensen — CHANGES_REQUESTED** [review]
Looks very good! I have made some suggestions for abstraction and redundant if statements

File: c-svc/transform.c (hunk 3 above)
Comment (humleflue) [reply c-svc/transform.c:11]: I think it is less clear what is happening. What do you think?
```
        for(j = 0; j < VECTOR_DIM; j++) {
            tso->observation[i+j] = newVector[j];
        }
```

File: c-svc/test/TransformTests.c (hunk 9 above)
Comment (humleflue) [reply c-svc/test/TransformTests.c:72]: Using the same functionality as what?

**Review by christopher-jensen — APPROVED** [review]

**Pull request merged**